src/test/unittest/ut_setup.h

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * ut_setup.h -- utility helper functions for libpmem2 tests setup
```

Comment:
I would probably just move all of this to `ut_pmem2_utils.h`.

Reply:
I described the problem above.

Reply:
Just don't use non-public API if it's causing linking problems.

Reply:
I created two files ut_setup.c and ut_setup_integration.c. These two approaches to testing (using nonpublic API and using only public API) are important IMO.